pkg/logql/syntax/ast_test.go

```diff
@@ -539,6 +581,10 @@ func TestStringer(t *testing.T) {
 			in:  `{app="foo"} !~ ip("127.0.0.1") or "foo"`,
 			out: `{app="foo"} !~ ip("127.0.0.1") !~ "foo"`,
 		},
+		{
+			in:  `{app="foo"} !> "<_> foo <_>" or "foo <_>" !> "foo <_> baz"`,
```

Comment:
I think the or is a bug here and should not be support but I see it elsewhere. I didn't know about this cc @kavirajk

Reply:
It looks legit to me: note the De Morgan's law a few lines above (`!(A || B) == !A && !B`):

`not match ("<_> foo <_>" or "foo <_>")` is the same as 
`not match "<_> foo <_>" and not match "foo <_>"`